server/src/main/java/org/opensearch/search/SearchModule.java

```diff
@@ -347,7 +343,6 @@ public SearchModule(Settings settings, List<SearchPlugin> plugins) {
         registerShapes();
         registerIntervalsSourceProviders();
         queryPhaseSearcher = registerQueryPhaseSearcher(plugins);
-        indexSearcherExecutorProvider = registerIndexSearcherExecutorProvider(plugins);
```

Comment:
Please revert this change: we exposed the extension point in the `SearchPlugin` and should respect that. However, if no executor is provided by the plugins, we could consult the feature flag and provide the thread pool executor.